scripts/restore.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Backup file directory path
+BACKUP_DIRECTORY_PATH="/palworld/backups"
+
+# Resotre path
+RESTORE_PATH="/palworld/Pal"
+
+term_error_handler() {
+    echo "An error occurred during server shutdown."
+    exit 1
+}
+
+restore_error_handler() {
+    printf "\033[0;31mAn error occurred during restore.\033[0m\n"
+    if [ -d "./tmp_save/Saved" ]; then
+      read -rp "I have a backup before recovery can proceed. Do you want to recovery it? (y/n): " RUN_ANSWER
+      if [[ $RUN_ANSWER == "y" ]] || [[ $RUN_ANSWER == "Y" ]]; then
+        rm -rf "$RESTORE_PATH/Saved"
+        mv "./tmp_save/Saved" "$RESTORE_PATH"
+        printf "\e[0;32mRecovery complete.\e[0m\n"
+      fi
+    fi
+
+    rm -rf "./tmp"
+    rm -rf "./tmp_save"
+
+    exit 1
+}
+
+if [ "${RCON_ENABLED}" != true ]; then
+  echo "RCON is not enabled. Please enable RCON to use this feature."
+  exit 1
+fi
+
+# Show up backup list
+echo "Backup List:"
+select BACKUP_FILE in "$BACKUP_DIRECTORY_PATH"/*; do
+    if [ -n "$BACKUP_FILE" ]; then
+        echo "Selected backup: $BACKUP_FILE"
+        break
+    else
+        echo "Invalid selection. Please try again."
+    fi
+done
+
+if [ -f "$BACKUP_FILE" ]; then
+  echo "Do you want to continue with the command?"
+  read -rp "When you run it, the server will be stopped and the recovery will proceed. (y/n): " RUN_ANSWER
+  if [[ $RUN_ANSWER == "y" ]] || [[ $RUN_ANSWER == "Y" ]]; then
+    printf "\e[0;32m*****STARTING PROCESS*****\e[0m\n"
+    # Shutdown server
+    trap 'term_error_handler' ERR
+
+    printf "\e[0;32m*****SHUTDOWN SERVER*****\e[0m\n"
+    if [ "${RCON_ENABLED}" = true ]; then
+        rcon-cli -c /home/steam/server/rcon.yaml save
+        rcon-cli -c /home/steam/server/rcon.yaml "shutdown 1"
+    else
+        echo "RCON is not enabled. Please enable RCON to use this feature."
+    fi
+    printf "\e[0;32mShutdown complete.\e[0m\n"
```

Comment:
You should exit the script if RCON is not enabled.
Maybe move the shutdown print statement so it is not displayed when you RCON is not enabled.
```suggestion
    if [ "${RCON_ENABLED}" = true ]; then
        printf "\e[0;32m*****SHUTDOWN SERVER*****\e[0m\n"
        rcon-cli -c /home/steam/server/rcon.yaml save
        rcon-cli -c /home/steam/server/rcon.yaml "shutdown 1"
    else
        echo "RCON is not enabled. Please enable RCON to use this feature. Unable to restore backup."
        exit 1
    fi
    printf "\e[0;32mShutdown complete.\e[0m\n"
```